api/src/main/java/edu/wpi/first/shuffleboard/api/widget/ComponentType.java

```diff
@@ -0,0 +1,17 @@
+package edu.wpi.first.shuffleboard.api.widget;
+
+import java.util.function.Supplier;
+
+/**
+ * A ComponentType is a Java object that encapsulates a single "variety" of component that the
```

Comment:
Why do programmers need to know that it is a Java object?

Reply:
Because it's distinct from a Java Type, which is what you'd normally expect this to be

Reply:
What is a Java Type?